### ISSUE TITLE
[Feature]: Widget deprecation notifications

### ISSUE THREAD

**Issue by riodeuno** [appsmithorg/appsmith#9014]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

A way to communicate with the Appsmith developers that a particular widget will be deprecated, subtly prompting them to upgrade the widget to the newly shipped version.

### Why should this be worked on?

We have seen on many occasions that making improvements in the DX of a widget, leads to breaking changes in the widget. In these scenarios, we create new versions of the widget to help Appsmith developers take advantage of the new features. Unfortunately, there exists no messaging today, in the platform, which informs Appsmith developers to if they're using an older version of the widget, which may not be maintained further into the future.

**Comment by zzgvh** [comment]
This is a much needed feature IMHO. I've had issues on numerous occasions that were resolved by deleting widgets and creating them anew. 

Once, a few months ago I had to rebuild a whole app, I think it was when cyclic dependency checks were added or more strictly enforced. I had quite a few widgets that referred to others and my impression is that when that change was added the cyclic deps were identified, but somehow incorrectly, maybe because the widgets were old. So the errors remained even after the offending widgets were removed. My only option was to rebuild the whole app.

So a way to "upgrade" widgets would be very useful!
